lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -447,6 +420,55 @@ void TypeLoweringVisitor::lowerBlock(Block *block) {
   }
 }
 
+ArrayAttr TypeLoweringVisitor::filterAnnotations(
+    MLIRContext *ctxt, ArrayAttr annotations, FIRRTLType srcType,
+    FlatBundleFieldEntry field, bool &hasDontTouch, StringAttr sym) {
+  hasDontTouch = false;
+  SmallVector<Attribute> retval;
+  if (!annotations || annotations.empty())
+    return ArrayAttr::get(ctxt, retval);
+  for (auto opAttr : annotations) {
+    if (auto subAnno = opAttr.dyn_cast<SubAnnotationAttr>()) {
+      // Apply annotations to all elements if fieldID is equal to zero.
+      if (subAnno.getFieldID() == 0) {
+        retval.push_back(subAnno.getAnnotations());
+        continue;
+      }
+
+      // Check whether the annotation falls into the range of the current field.
+      if (subAnno.getFieldID() >= field.fieldID &&
+          subAnno.getFieldID() <= field.fieldID + field.type.getMaxFieldID()) {
```

Comment:
Nit: This may be cleaner to invert the condition and continue instead of a large `if` block.